File: SingleNumber.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        temp = []  # 요소를 저장, 같은 값의 요소가 있으면 제거할거임
        for num in nums:
            if num in temp:
                temp.remove(num)
            else:
                temp.append(num)

        return temp[0]

File: SingleNumber.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int res = 0;
        for(int i=0; i<nums.size();  i++){
            res ^= nums[i];
        }
        return res;
    }
};

// xor연산을해서 계속 누적.
// 같은게 나온다면 xor연산으로 없어지겠지? 같은건 0으로 나오니까!